sbi/neural_nets/estimators/flowmatching_estimator.py

```diff
@@ -91,45 +87,46 @@ def __init__(
     def embedding_net(self):
         return self._embedding_net
 
-    def forward(self, input: Tensor, condition: Tensor, t: Tensor) -> Tensor:
-        """
-        Forward pass of the FlowMatchingEstimator.
+    def forward(self, input: Tensor, condition: Tensor, time: Tensor) -> Tensor:
+        """Forward pass of the FlowMatchingEstimator.
 
         Args:
-            input: The input tensor.
-            condition: The condition tensor.
-            t: The time tensor.
+            input: Original data, x0. (input_batch_shape, *input_shape)
+            condition: Conditioning variable. (condition_batch_shape, *condition_shape)
+            time: SDE time variable in [0,1].
 
         Returns:
             The estimated vector field.
         """
-        # temporal fix that will be removed when the nn builders are updated
-        t = self._get_temporal_t_shape_fix(t)
-
+        # Continue with standard processing (broadcast shapes etc.)
+        batch_shape_input = input.shape[: -len(self.input_shape)]
+        batch_shape_cond = condition.shape[: -len(self.condition_shape)]
         batch_shape = torch.broadcast_shapes(
-            input.shape[: -len(self.input_shape)],
-            condition.shape[: -len(self.condition_shape)],
+            batch_shape_input,
+            batch_shape_cond,
         )
 
-        input = torch.broadcast_to(input, batch_shape + self.input_shape)
-        condition = torch.broadcast_to(condition, batch_shape + self.condition_shape)
-        t = torch.broadcast_to(t, batch_shape + t.shape[1:])
+        # embed the conditioning variable
+        condition_emb = self._embedding_net(condition)
 
```

Comment:
We should enable z-scoring for flow matching estimators. One way to do it would be to match what we do for score estimators, i.e. define the marginal means and sds and do time-dependent z-scoring. This would then implicitly assume independence in the dimension, but I guess it's a start. Otherwise,  we need to add a separate normalizing of the input here, followed by rescaling the output in the end. But I think if we don't do any z-scoring at all and expect the users to do it then we are likely to get a lot of "flow-matching doesn't work well" questions.

Reply:
Moved to #1623